common/src/main/java/com/wynntils/core/storage/StorageManager.java

```diff
@@ -65,46 +65,25 @@ public void initFeatures() {
-        String baseName = storageable.getStorageJsonName();
-
-        Field[] annotatedStorages = FieldUtils.getFieldsWithAnnotation(storageable.getClass(), RegisterStorage.class);
-        for (Field field : annotatedStorages) {
-            try {
-                Object fieldValue = FieldUtils.readField(field, storageable, true);
-                if (!(fieldValue instanceof Storage<?>)) {
-                    throw new RuntimeException(
-                            "A non-Storage class was marked with @RegisterStorage annotation: " + field);
-                }
-            } catch (IllegalAccessException e) {
-                throw new RuntimeException("Failed to read @RegisterStorage annotated field: " + field, e);
-            }
-        }
-        List<Field> fields = FieldUtils.getAllFieldsList(storageable.getClass());
-        List<Field> storageFields =
-                fields.stream().filter(f -> f.getType().equals(Storage.class)).toList();
-
-        for (Field storageField : storageFields) {
-            try {
-                RegisterStorage storageInfo = Arrays.stream(annotatedStorages)
-                        .filter(f -> f.equals(storageField))
-                        .findFirst()
-                        .map(f -> f.getAnnotation(RegisterStorage.class))
-                        .orElse(null);
-                if (storageInfo == null) {
-                    throw new RuntimeException("A Storage is missing @RegisterStorage annotation:" + storageField);
-                }
-
-                Storage<?> storage = (Storage<?>) FieldUtils.readField(storageField, storageable, true);
-                String jsonName = baseName + "." + storageField.getName();
-                storages.put(jsonName, storage);
-
-                Type valueType = Managers.Json.getJsonValueType(storageField);
-                storageTypes.put(storage, valueType);
-                storageOwner.put(storage, storageable);
-            } catch (IllegalAccessException e) {
-                throw new RuntimeException(e);
-            }
+    public void registerStorageable(Storageable owner) {
+        Managers.Persisted.verifyAnnotations(owner);
+
+        Managers.Persisted.getPersisted(owner, Storage.class).stream()
+                .forEach(p -> processStorage(owner, p.a(), p.b()));
+    }
+
+    private void processStorage(Storageable owner, Field field, Persisted annotation) {
+        try {
+            String baseName = owner.getStorageJsonName();
+            Storage<?> storage = (Storage<?>) FieldUtils.readField(field, owner, true);
+            String jsonName = baseName + "." + field.getName();
+            storages.put(jsonName, storage);
+
+            Type valueType = Managers.Json.getJsonValueType(field);
+            storageTypes.put(storage, valueType);
+            storageOwner.put(storage, owner);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
```

Comment:
Is this the time to start catching these somewhere? Currently we would crash the system just like we used to..

Reply:
Perhaps. The illegal access would only happen if we were not allowed to read the field despite the `force` flag, I think. Which could only happen if we have a securit manager, I think. So it is more of a "won't happen but we need to have this by formality".

Anyway, this is a pure refactoring. Adding extra stability will have to come later.